### ISSUE TITLE
requiring atom-space-pen-views gives me something else

### ISSUE THREAD

**Issue by noseglid** [atom/atom-space-pen-views#5]
I'm having an issue where my package (atom-build) is trying to extend the `View` provided by `space-pen`.

As per [this document](https://github.com/atom/atom/pull/4169) I've changed it to require 'atom-space-pen-views' instead, then taking `View` from the export.

However, if atom-build is installed as a normal package (e.g. `apm install atom-build`), it wont work. It appears as `View` is undefined. Weird.

I asked about it on [atom discuss board](https://discuss.atom.io/t/requiring-atom-space-pen-views-give-me-a-filter-function/14101/2) and @postcasio found that what I am getting in return from `require('atom-space-pen-views')` is actually what another dependency of mine (`ansi-to-html`) is exporting.

Something is terribly wrong here. I'm not sure this issue actually belongs on this project, but I'm posting it here since this is the one acting up.

**Comment by kevinsawicki** [comment]
Is this happening with the `0.22.3` of your build package?

**Comment by noseglid** [comment]
Yes. Looks like most users are having the issue so simply installing it should cause the problem.

**Comment by noseglid** [comment]
So, looking at the `package.json` file in `~/.atom/packages/build/` the error is obvious:

This key: `_atomModuleCache.dependencies` have an entry for atom-space-pen-views. The `path` element in it is set to: `node_modules/ansi-to-html/lib/ansi_to_html.js`. Changing it to:  `node_modules/atom-space-pen-views/lib/main.js` and everything works smoothly.

Now who puts all that metadata there. apm ?

**Comment by kevinsawicki** [comment]
Yeah, apm does that, taking a look now

**Comment by kevinsawicki** [comment]
Fixed by https://github.com/atom/atom-space-pen-views/commit/e3beeb4364a58d77ad5e4554483c15263c0935c6

There were actually two issues going on here.

There was a definite but in the module cache builder where the main path for a previous module was being used when the main path for the current module couldn't be resolved.  This has been fixed on atom/atom master and will be included in the next Atom release, 0.167.

The second issue was that this module wasn't precompiling the CoffeeScript files to JavaScript.  This is why the main path wasn't resolve-able. I've fixed that in https://github.com/atom/atom-space-pen-views/commit/e3beeb4364a58d77ad5e4554483c15263c0935c6 and published a 2.0.3 version.

If you upgrade to 2.0.3 your package should now be fixed, I've opened up a PR, https://github.com/noseglid/atom-build/pull/30

So if you release after upgrading, people should get into a working state again without having to upgrade Atom.
